packages/neo-one-node-vm/src/syscalls.ts

```diff
@@ -1120,58 +1127,111 @@ export const SYSCALLS: { readonly [K in SysCallEnum]: CreateSysCall } = {
   'System.Contract.Update': getContractFee((argsIn, fee) =>
     createSysCall({
       name: 'System.Contract.Update',
-      in: 9,
-      out: 1,
+      in: 2,
       fee,
       invoke: async (options) => {
-        if (options.context.init.triggerType !== TriggerType.Application) {
-          throw new InvalidVerifySyscallError(options.context, 'Neo.Contract.Update');
-        }
         const { context: contextIn, args } = options;
         let context = contextIn;
-        const { contract, created } = await createContract({ context, args });
-        if (contract.hasStorage && created) {
-          await context.blockchain.storageItem
-            .getAll$({
-              hash: context.scriptHash,
-            })
-            .pipe(
-              concatMap((item) =>
-                defer(async () =>
-                  context.blockchain.storageItem.add(
-                    new StorageItem({
-                      hash: contract.hash,
-                      key: item.key,
-                      value: item.value,
-                      flags: StorageFlags.None,
-                    }),
+        const { hash, script, manifestString } = validateContractArgs({ context, args });
+
+        let contract = await context.blockchain.contract.tryGet({ hash: context.scriptHash });
+        if (contract === undefined) {
+          throw new ContractHashNotFoundError(contextIn, context.scriptHash);
+        }
+
+        if (script.length > 0) {
+          const tryHash = await context.blockchain.contract.tryGet({ hash });
+          if (hash.equals(context.scriptHash) || tryHash !== undefined) {
+            throw new ContractAlreadyDeployedError(context, hash);
+          }
+          contract = new Contract({
+            script,
+            manifest: new ContractManifest({
+              abi: new ContractABI({
+                hash,
+                entryPoint: contract.manifest.abi.entryPoint,
+                methods: contract.manifest.abi.methods,
+                events: contract.manifest.abi.events,
+              }),
+              groups: contract.manifest.groups,
+              features: contract.manifest.features,
+              permissions: contract.manifest.permissions,
+              trusts: contract.manifest.trusts,
+              safeMethods: contract.manifest.safeMethods,
+              extra: contract.manifest.extra,
+            }),
+          });
+
+          await context.blockchain.contract.add(contract);
+
+          if (contract.manifest.hasStorage) {
+            await context.blockchain.storageItem
+              .getAll$({ hash: context.scriptHash })
```

Comment:
Change this to check if any storage items exist.  Maybe use a pattern similar to `Neo.Storage.Find` with `toArray` or make sure of the `count` operator: https://rxjs-dev.firebaseapp.com/api/operators/count.